src/main/java/Duke.java

```diff
@@ -1,10 +1,93 @@
+import java.util.Scanner;
+import java.time.LocalDate;
+
+/**
+ * Represent the main driving class Duke.
+ */
 public class Duke {
+
+    protected Ui ui;
+    protected Storage storage;
+    protected Tasklist tasklist;
+
+    /**
+     * Constructor of Duke.
+     * Initialise Ui, Storage and Tasklist
+     * @param filepath of the file to be used in storage
+     */
+    public Duke(String filepath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filepath);
+        this.tasklist = new Tasklist(storage.readFile());
+    }
+
+    /**
+     * The main logic of the chatbot.
+     */
+    public void run() {
+        this.ui.printIntro();
+        Scanner myScanner = new Scanner(System.in);
+
+        while (true) {
+            String word = myScanner.nextLine();
+            try {
+                String[] parsed = TextParser.myFirstParser(word);
+                String keyword = parsed[0];
+                if (keyword.equals("bye")) {
```

Comment:
It would be good if you add an empty line at the end of each block. It spaces out the code so it's not so hard to read.